lineapy/graph_reader/session_artifacts.py

```diff
@@ -308,6 +321,11 @@ def _slice_session_artifacts(self) -> None:
                     pred_nodes
                 )
 
+                use_cache = (
```

Comment:
Ok, I think this name actually deserves to be something like `node_collection_use_cache`, since it has a value if the node collection is associated with a "cached artifact" and is None otherwise. 

Right now it has the same name as use_cache for SessionArtifact and ArtifactCollection. For those two classes use_cache contains a list of artifacts to use as input. However, here `use_cache` is used for NodeCollection and the one node `n` which defines the artifact of the NodeCollection.